pkg/common/cns-lib/volume/listview.go

```diff
@@ -181,13 +205,24 @@ func (l *ListViewImpl) AddTask(ctx context.Context, taskMoRef types.ManagedObjec
 // RemoveTask removes task from listview and the internal map
 func (l *ListViewImpl) RemoveTask(ctx context.Context, taskMoRef types.ManagedObjectReference) error {
 	log := logger.GetLogger(ctx)
+	deadline, ok := ctx.Deadline()
+	if !ok || time.Now().After(deadline) {
+		log.Infof("op timeout. context deadline exceeded. using listview context without a timeout")
+		ctx = l.ctx
```

Comment:
Can you explain why we need to replace method ctx? and how l.ctx is ensured it is not timed out?

Reply:
Added comment